File: Demo/SettingsViewController.m
//
//  SettingsViewController.m
//  Springback
//
//  Created by Sheun  Olatunbosun on 10/22/16.
//  Copyright © 2016 Sheun  Olatunbosun. All rights reserved.
//

#import "SettingsViewController.h"

@interface SettingsViewController ()
{
    IBOutlet UISwitch *controlEnabledSwitch;
    IBOutlet UISwitch *showKnobsSwitch;
    IBOutlet UISwitch *reversePanSwitch;
    IBOutlet UISwitch *boundaryLimitSwitch;
}
@end

@implementation SettingsViewController

- (instancetype)init
{
    UIStoryboard *storyboard = [UIStoryboard storyboardWithName:@"Main" bundle:nil];
    return [storyboard instantiateViewControllerWithIdentifier:@"Settings"];
}

- (void)viewDidLoad {
    [super viewDidLoad];
    
    controlEnabledSwitch.on = self.controlEnabled;
    showKnobsSwitch.on = self.showKnobs;
    reversePanSwitch.on = self.reversePan;
    boundaryLimitSwitch.on = self.boundaryLimit;
}

- (IBAction)switchClicked:(id)sender
{
    UISwitch *switchControl = sender;
    
    if (switchControl == controlEnabledSwitch) self.controlEnabled = switchControl.isOn;
    
    if (switchControl == showKnobsSwitch) self.showKnobs = switchControl.isOn;
    
    if (switchControl == reversePanSwitch) self.reversePan = switchControl.isOn;
    
    if (switchControl == boundaryLimitSwitch) self.boundaryLimit = switchControl.isOn;
}
@end

File: DemoSwift/SettingsViewController.swift
//
//  SettingsController.swift
//  Springback
//
//  Created by sodev on 7/8/17.
//  Copyright © 2017 Sheun  Olatunbosun. All rights reserved.
//

import UIKit

class SettingsViewController: UIViewController {
    
    @IBOutlet var controlEnabledSwitch: UISwitch!
    @IBOutlet var showKnobsSwitch: UISwitch!
    @IBOutlet var reversePanSwitch: UISwitch!
    @IBOutlet var boundaryLimitSwitch: UISwitch!
    
    public var controlEnabled: Bool = true
    public var showKnobs: Bool = true
    public var reversePan: Bool = false
    public var boundaryLimit: Bool = true
    
    static func create() -> SettingsViewController {
        let sb = UIStoryboard(name:"Main", bundle:nil)
        return sb.instantiateViewController(withIdentifier: "Settings") as! SettingsViewController
    }
    
    @IBAction func switchClicked(_ sender: Any?) {
        if let sender = sender as? UISwitch{
            switch sender {
            case controlEnabledSwitch:
                controlEnabled = sender.isOn
                break
            case showKnobsSwitch:
                showKnobs = sender.isOn
                break
            case reversePanSwitch:
                reversePan = sender.isOn
                break
            case boundaryLimitSwitch:
                boundaryLimit = sender.isOn
                break
            default:
                break
            }
        }
    }
    // MARK: ViewController
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        controlEnabledSwitch.isOn = self.controlEnabled
        showKnobsSwitch.isOn = self.showKnobs
        reversePanSwitch.isOn = self.reversePan
        boundaryLimitSwitch.isOn = self.boundaryLimit
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
}
